src/udiskslinuxpartitiontable.c

```diff
@@ -601,13 +603,30 @@ udisks_linux_partition_table_handle_create_partition (UDisksPartitionTable   *ta
       g_free (escaped_name);
 #else
       size = end_bytes - (start_mib * MIB_SIZE); /* recalculate size with the new end_bytes */
-      if (! bd_part_create_part (device_name, BD_PART_TYPE_REQ_NORMAL, start_mib * MIB_SIZE,
-                                 size, BD_PART_ALIGN_OPTIMAL, &error))
+      part_spec = bd_part_create_part (device_name, BD_PART_TYPE_REQ_NORMAL, start_mib * MIB_SIZE,
+                                       size, BD_PART_ALIGN_OPTIMAL, &error);
+      if (! part_spec)
         {
           g_dbus_method_invocation_take_error (invocation, error);
           goto out;
         }
 
+      /* set name if given */
+      if (strlen (name) > 0)
+        {
+          escaped_name = udisks_daemon_util_escape (name);
+          if (! bd_part_set_part_name (device_name, part_spec->path, escaped_name, &error))
```

Comment:
libblockdev doesn't use shell to set partition name, so the name shouldn't be escaped at all.